src/mixer/main.go

```diff
@@ -224,6 +226,69 @@ func cmdAddBundles(args []string) {
 	b.AddBundles(bundles, *force, *git)
 }
 
+func cmdInit(args []string) {
+	initcmd := flag.NewFlagSet("init", flag.ExitOnError)
+	confOnlyflag := initcmd.Bool("config-only", false, "Only initialize the builder config")
+	allflag := initcmd.Bool("all", false, "Create a mix with all Clear bundles included")
+	clearflag := initcmd.Int("clearver", 0, "Supply the Clear version to compose the mix from")
+	mixflag := initcmd.Int("mixver", 0, "Supply the Mix version to build")
+	initcmd.Parse(args)
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		helpers.PrintError(err)
+		os.Exit(1)
+	}
+
+	err = os.Mkdir(pwd+"/rpms", 0755)
+	if os.IsExist(err) {
+		fmt.Println("Directory '/rpms' already exists. Skipping...")
+	} else if err != nil {
+		helpers.PrintError(err)
+		os.Exit(1)
+	}
+
+	err = os.Mkdir(pwd+"/local", 0755)
+	if os.IsExist(err) {
+		fmt.Println("Directory '/local' already exists. Skipping...")
+	} else if err != nil {
+		helpers.PrintError(err)
+		os.Exit(1)
+	}
+
+	err = helpers.CopyFile(pwd+"/builder.conf", "/usr/share/defaults/bundle-chroot-builder/builder.conf", false)
+	if os.IsExist(err) {
+		fmt.Println("File 'builder.conf' already exists. Skipping...")
+	} else if err != nil {
+		helpers.PrintError(err)
+		os.Exit(1)
+	} else {
+		raw, err := ioutil.ReadFile(pwd + "/builder.conf")
+		if err != nil {
+			helpers.PrintError(err)
+			os.Exit(1)
+		}
+
+		data := strings.Replace(string(raw), "/home/clr/mix", pwd, -1)
+		data += "\nRPMDIR=" + pwd + "/rpms\n"
+		data += "REPODIR=" + pwd + "/local\n"
```

Comment:
This might not do what you want it to.  This will append it to the end of the file, but that will put it in the [Server] section of the TOML* file.

\* Technically it's not TOML, it's INI. The file exists for the bundle-chroot-builder, which is written in Python and uses [the configparser library](https://docs.python.org/3/library/configparser.html), which only handles INI not TOML.

Reply:
Right now, at least for mixer, the sections are just for show. The "use mixer tool" documentation also don't specify where to put them, so they will usually end up at the end of the file. But I agree, that's not the best place to have it. My plan was to handle it properly while reworking the config parser.

Reply:
They are "for show" for mixer right now, but the configparse library used by bundle-chroot-builder actually does separate things out by section correctly.

The [Server] section is parsed out [here](https://github.com/clearlinux/bundle-chroot-builder/blob/master/bundle-chroot-builder.py#L205) and used [here](https://github.com/clearlinux/bundle-chroot-builder/blob/master/bundle-chroot-builder.py#L182). Based on how they're using it, these new values would be ignored and not hurt anything, but it's a dangerous coupling that results from two different tools relying on the same config file.

While I agree this should be "done properly" in a config file rewrite, for now you could insert it in to the right section by finding the location of "\n\n[swupd]" and inserting them before that.

Reply:
... though, I guess REPODIR and RPMDIR are only used by mixer, so there isn't really a "right section". Maybe it should go in its own [Mixer] section?

Reply:
@mdhorn @kevincwells  would it be useful to have a --prefix or --path option to this so you can mixer init --path /some/diff/path/than/currdir? Probably out of scope for this PR but good to think about for future improvement.

Reply:
@tmarcu I think that was originally part of the proposal @rchiossi had suggested. I can see uses for that, but I'd be fine with it being a future improvement.

Reply:
I agree for future, just wanted to mention it.